docs/extend-graphql-schema.md

```diff
@@ -3,4 +3,568 @@ id: extend-graphql-schema
+      "state": "Ile-de-France",
+      "country": "France",
+      "postcode": "75001",
+      "country_code": "fr"
+    }
+  }
+  // …
+]
+```
+
+The schema defined earlier specified that `physicalStores` will return a list of
+`PhysicalStore` objects, which do not match the format returned by the remote
+service so far.
+
+```
+type Query {
+  "Our physical stores"
+  physicalStores: [PhysicalStore]
+}
+```
+
+In this implementation, we chose to use resolvers for converting data from the
+remote service into an object matching the `PhysicalStore` type definition. This
+time, by adding a resolver for each of its two fields:
+
+```js
+export default {
+  // …
+  PhysicalStore: {
+    coordinates: store => ({
+      latitude: store.lat,
+      longitude: store.lon
+    }),
+    openingHours: () => generalOpeningHours
+  }
+  // …
+};
+```
+
+The `coordinates` field is resolved by using the initial `store` information
+(first argument) and converting them into a structure matching our Schema. With
+simplification in mind, we implemented `openingHours` field resolution in a
+naive way: by supposing that all stores shared the same established opening
+hours (defined in the `generalOpeningHours` constant).
+
+> One may combine these two approaches to fetch openingHours from another
+> service, using for instance the `place_id` to access a REST detail endpoint.
+> It could be a good thing to try if you want to experiment and start grasping
+> the power behind a GraphQL middleware in your architecture!
+
+You might now be able to understand the rest of this resolver map, which consist
+of data transformation for the `PostalAddress.street` field (data returned by
+the remote service match the `PostalAddress` type definition for the other
+fields) and `Time` representations.
+
+```js
+export default {
+  // …
+  Time: {
+    h: time => time.split(":")[0],
+    m: time => time.split(":")[1]
+  },
+
+  PostalAddress: {
+    street: address => address.road
+  }
+};
+```
+
+> ProTip™: you can debug the data passed in a resolver using `console.log` to
+> view its shape in your server console shell output. To do so, a convenient
+> thing to know is that `console.log` returns a falsy value. Hence, in the
+> resolver above you could debug the `time` param by updating the resolver as
+> below:
+>
+> `h: time => console.log(time) || time.split(":")[0],`
+
+### Refactor to context and loaders
+
+We have not analyzed yet the `generalOpeningHours` constant and `findAllStores`
+function called by resolvers, for a simple reason: it is not really relevant to
+understand how Front-Commerce works.
+
+This part of the code is pure EcmaScript and is agnostic to the
+library/framework/otherTrendyThing used. It means that if you have a legacy node
+codebase you may be able to reuse some of your code, and if you want to abandon
+GraphQL or Front-Commerce you could still use this code.
+
+For this reason **we recommend to keep as much as your code in pure EcmaScript
```

Comment:
That's not only about EcmaScript, it's about extracting the logic from the resolvers into independant functions.

Reply:
I'm not sure `as much as ` is correct here?

It is a bit convoluted but I would phrase it `as much of your code as possible in pure ECMAScript `